devices/RobotDynamicsEstimatorDevice/src/RobotDynamicsEstimatorDevice.cpp

```diff
@@ -235,6 +307,16 @@ bool RobotDynamicsEstimatorDevice::resizeEstimatorInitialState(
 
 bool RobotDynamicsEstimatorDevice::setEstimatorInitialState()
 {
+    // Set logPrefix
+    const std::string logPrefix = "[RobotDynamicsEstimatorDevice::setEstimatorInitialState]";
+
+    iDynTree::Model model = m_kinDyn->model();
+    // if (!model.isValid())
+    // {
+    //     log()->error("{} The model is not valid.", logPrefix);
+    //     return false;
+    // }
```

Comment:
Why was this removed?

Reply:
As we discussed f2f, when we create the submodels, something checked by the `isValid` function is no longer respected, even if the submodel seems ok (I printed it on the terminal with the `toString` function). I did not investigate what exactly, but I suspect this is related to the sequence of link indices, as all the links not specified in the list of links to consider for the submodel will be deleted.

Reply:
Maybe, I can try to check on a simple test what isValid is checking and what is broken in the submodel generated by iDynTree. Perhaps it is easier to test using the iDynTree tests.

Reply:
I agree. The fact that we have no-valid model scares me 😱

Reply:
Hi @GiulioRomualdi, I have added back this check. This will be tested on the robot. By debugging the tests, I did not find any problem. We may merge the PR, and if I find any problem with the robot, I'll debug it.